### DISCUSSION TITLE
Select multiple condition on single field in single table

### DISCUSSION THREAD

**Discussion by lakshit-shah-95** [prisma/prisma#16083] [Q&A] [answered]
Hey, I want to fetch people based on conditions provided. 
My postgres raw query goes like this `SELECT * FROM person WHERE ((age > 15 AND age < 20) OR age > 60) OR ((height > 3 AND height < 2) OR height > 6).  I am fine with not using between operator. person table is this

```
model person {
  id             String     @id
  name           String
  age            Int      
  height         Float
}
```

Thanks

**Comment by nurul3101** [comment, marked as answer]
Hey @lakshit-shah-95 👋 

Can you try this query? This should satisfy the conditions that you have specified.
This could look complicated, but the key is to unwrap the parenthesis of where clause one by one.

```typescript
  await prisma.person.findMany({
    where: {
      OR: [
        {
          OR: [
            {
              AND: [
                {
                  age: {
                    gt: 15,
                  },
                },
                {
                  age: {
                    lt: 20,
                  },
                },
              ],
            },
            {
              age: {
                gt: 60,
              },
            },
          ],
        },
        {
          OR: [
            {
              AND: [
                {
                  height: {
                    gt: 3,
                  },
                },
                {
                  height: {
                    lt: 2,
                  },
                },
              ],
            },
            {
              height: {
                gt: 6,
              },
            },
          ],
        },
      ],
    },
  });
```

Here's a reference to [AND](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#and) and [OR](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#or) operators.
Do let me know in case you face any other issues.

**Reply by lakshit-shah-95** [reply]
Thanks for the answer @nurul3101 . It works.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
